pkg/volume/emptydir/quota_test.go

```diff
@@ -247,3 +253,82 @@ func TestApplyQuota(t *testing.T) {
 		}
 	}
 }
+
+func TestIsMountedWithGrpquota(t *testing.T) {
+
+	// Substitute in the actual mount line we're after for each test:
+	mountOutput := `/dev/mapper/fedora_system-root on / type ext4 (rw,relatime,seclabel,data=ordered)
+selinuxfs on /sys/fs/selinux type selinuxfs (rw,relatime)
+systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=26,pgrp=1,timeout=0,minproto=5,maxproto=5,direct)
+debugfs on /sys/kernel/debug type debugfs (rw,relatime,seclabel)
+mqueue on /dev/mqueue type mqueue (rw,relatime,seclabel)
+hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,seclabel)
+tmpfs on /tmp type tmpfs (rw,seclabel)
+%s
+nfsd on /proc/fs/nfsd type nfsd (rw,relatime)
+/dev/sda1 on /boot type ext4 (rw,relatime,seclabel,data=ordered)
+/dev/mapper/fedora_system-home on /home type ext4 (rw,relatime,seclabel,data=ordered)
+/dev/sdb1 on /storage type btrfs (rw,relatime,seclabel,space_cache,subvolid=5,subvol=/)
+`
+	var fsDevice = "/dev/mapper/openshift--vol--dir"
+	var volumeDir = "/var/lib/origin/openshift.local.volumes"
+
+	tests := map[string]struct {
+		MountLine string
+		ExpResult bool
+		ExpError  string // sub-string to be searched for in error message
+	}{
+		"grpquota": {
+			MountLine: fmt.Sprintf("%s on %s type xfs (rw,relatime,seclabel,attr2,inode64,grpquota)", fsDevice, volumeDir),
+			ExpResult: true,
+		},
+		"gquota": {
+			// May not be possible in the real world (would show up as grpquota) but just in case:
+			MountLine: fmt.Sprintf("%s on %s type xfs (rw,relatime,seclabel,attr2,inode64,gquota)", fsDevice, volumeDir),
+			ExpResult: true,
+		},
+		"gqnoenforce": {
+			MountLine: fmt.Sprintf("%s on %s type xfs (rw,relatime,seclabel,attr2,inode64,gqnoenforce)", fsDevice, volumeDir),
+			ExpResult: false,
+		},
+		"noquota": {
+			MountLine: fmt.Sprintf("%s on %s type xfs (rw,relatime,seclabel,attr2,inode64,noquota)", fsDevice, volumeDir),
+			ExpResult: false,
+		},
+		"device not in output": {
+			MountLine: fmt.Sprintf("/dev/sdb1 on %s type xfs (rw,relatime,seclabel,attr2,inode64,noquota)", volumeDir),
+			ExpResult: false,
+			ExpError:  "unable to find device in mount output",
+		},
+	}
+
+	for name, test := range tests {
+		t.Logf("running TestIsMountedWithGrpquota: %s", name)
+
+		mockCmdRunner := &mockQuotaCommandRunner{}
+		mockCmdRunner.RunMountOptionsCommandResponse = &cmdResponse{
+			Stdout: fmt.Sprintf(mountOutput, test.MountLine),
+			Stderr: "",
+			Error:  nil,
+		}
+		mockCmdRunner.RunFSDeviceCommandResponse = &cmdResponse{
+			Stdout: fmt.Sprintf("Filesystem\n%s", fsDevice),
+			Stderr: "",
+			Error:  nil,
+		}
+
+		mountedWithQuota, err := isMountedWithGrpquota(mockCmdRunner, volumeDir)
+		if test.ExpError != "" {
+			if err == nil {
+				t.Errorf("failed: '%s', expected error but got none", name)
```

Comment:
Use `%q` to quote strings.

No need to print `failed` since you're indicating a failure via the call to `t.Errorf`